tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,184 @@
+    script_run("kubectl version --client");
+}
+
+sub install_aistack_chart {
+    my ($instance, $ai_chart_repo, $namespace, $vf_name) = @_;
+    record_info('AISTACK charts install');
+    my $SECRET_application_collection = get_var('_SECRET_DOCKER');
+    my $cert_repo = get_var('HELM_CERTS');
+    my $docker_user_name = 'yarunachalam@suse.com';
+    my $repo_url = 'github.com/SUSE/private-ai-charts';
+    # my $repo_url = get_var('HELM_CHARTS');
+    my $git_token = get_var('_SECRET_SSH');
+
+    # config kubectl
+    assert_script_run("mkdir -p ~/.kube");
+    assert_script_run("sudo cp /etc/rancher/rke2/rke2.yaml ~/.kube/config");
+    assert_script_run("kubectl config get-contexts");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config view");
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manger --create-namespace --version v1.15.2 --set crds.enabled=true", timeout => 120);
+
+    sleep 180;
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns $namespace");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$SECRET_application_collection' -n $namespace", timeout => 120);
+
+    # Install private-ai-stack
+    my $gitlab_clone_url = 'https://git:' . $git_token . '@' . $repo_url;
+    assert_script_run("git clone $gitlab_clone_url");
+    assert_script_run("curl -o $vf_name $ai_chart_repo", timeout => 120);
+    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml", timeout => 120);
+    assert_script_run("helm upgrade --install suse-private-ai private-ai-charts --namespace $namespace --create-namespace --values $vf_name", timeout => 300);
+    assert_script_run("helm list --namespace $namespace");
+
+    # Check pod status and log for successful
+    my @out = split(' ', script_output("kubectl get pods --namespace $namespace -o custom-columns=':metadata.name'"));
+    record_info("Pod names", join(" ", @out));
+    foreach my $pod (@out) {
+        my $counter = 0;
+        while ($counter++ < 30) {
+            sleep 10;
+            my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
+            if ($status eq 'Running') {
+                record_info("$pod is running. ");
+                next;
+            }
+            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
```

Comment:
I think no need of another variable, naming it once i.e. `my $output` for both uses.

Reply:
Also add comments what each if-block doing

Reply:
Code changed.